strawberry/type.py

```diff
@@ -33,10 +64,12 @@ def _process_type(
 ):
     name = name or to_camel_case(cls.__name__)
 
-    wrapped = dataclasses.dataclass(cls)
+    wrapped = _wrap_dataclass(cls)
 
     interfaces = _get_interfaces(wrapped)
 
+    # TODO: check for missing types here
```

Comment:
@BryceBeagle what if we add some safety checks here? At least to check if all fields have return types. Would be amazing to check that before we do the schema generation, as the schema generation shouldn't have to deal with missing types :)